src/components/product/product.service.ts

```diff
@@ -441,27 +401,39 @@ export class ProductService {
         value: securedProperties.purposes.value ?? [],
       },
       produces: {
+        ...(securedProperties.scriptureReferencesOverride
+          ? securedProperties.scriptureReferencesOverride
+          : {
+              canRead: false,
+              canEdit: false,
+            }),
         value: {
           ...producible,
-          id: produces.p.properties.id,
+          id: pr.p.properties.id,
           __typename: (ProducibleType as any)[typeName],
-          scriptureReferences: !scriptureReferencesOverride.length
+          scriptureReferences: !scriptureReferencesOverrideValue.length
             ? producible?.scriptureReferences
             : {
-                canRead: result.canScriptureReferencesOverrideRead,
-                canEdit: result.canScriptureReferencesOverrideEdit,
-                value: scriptureReferencesOverride,
+                ...(securedProperties.scriptureReferencesOverride
+                  ? securedProperties.scriptureReferencesOverride
+                  : {
+                      canRead: false,
+                      canEdit: false,
+                    }),
+                value: scriptureReferencesOverrideValue,
               },
         },
-        canRead: result.canProducesRead,
-        canEdit: result.canProducesEdit,
       },
       scriptureReferencesOverride: {
-        canRead: result.canScriptureReferencesOverrideRead,
-        canEdit: result.canScriptureReferencesOverrideEdit,
-        value: !scriptureReferencesOverride.length
+        ...(securedProperties.scriptureReferencesOverride
+          ? securedProperties.scriptureReferencesOverride
+          : {
+              canRead: false,
+              canEdit: false,
+            }),
```

Comment:
scriptureReferencesOverride should always be defined and have default values if needed.

Reply:
https://i.imgur.com/8kTFDK8.png

Reply:
scriptureReferencesOverride and produces have undefined type as union.

Reply:
Ah. That's from the usage of `AnyProduct`. I would create a type alias
```tsx
type ProductDbProps = DbPropsOfDto<DirectScriptureProduct & DerivativeScriptureProduct>;
```